crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs

```diff
@@ -78,18 +80,51 @@ impl Violation for YodaConditions {
     }
 }
 
-/// Return `true` if an [`Expr`] is a constant or a constant-like name.
-fn is_constant_like(expr: &Expr) -> bool {
+/// Comparisons left hand side must not be more [`ConstantLikelihood`] than their right hand side
+#[derive(PartialEq, Eq, PartialOrd, Ord, Debug)]
+enum ConstantLikelihood {
+    Unlikely = 0,
+    Probably = 1,   // CAMEL_CASED vars
+    Definitely = 2, // literals, empty dicts and tuples...
+}
+
+fn how_likely_constant_str(s: &str) -> ConstantLikelihood {
+    if str::is_cased_uppercase(s) {
+        ConstantLikelihood::Probably
+    } else {
+        ConstantLikelihood::Unlikely
+    }
+}
+
+/// Return [`Expr`] [`ConstantLikelihood`] level depending on simple heuristics.
+fn how_likely_constant(expr: &Expr) -> ConstantLikelihood {
+    if expr.is_literal_expr() {
+        return ConstantLikelihood::Definitely;
+    }
```

Comment:
Entirely personal style but I'd probably just put this in the `match` with all the cases rather than adding an early return. I just find it a bit clearer.

Reply:
Feel free to correct me, I'm here to learn 🙂 
Would you say this is idiomatic: https://github.com/astral-sh/ruff/blob/7879f9e921c3a67f1b89c290e591bb01799b5479/crates/ruff_linter/src/rules/flake8_simplify/rules/yoda_conditions.rs#L98-L100
(trying to DRY this invocation from multiple patterns)

Reply:
Yeah, I think what you have here with the `match` is a nice approach.